src/droid/handler.clj

```diff
@@ -32,13 +34,19 @@
        ;; Merge the hashmaps corresponding to each environment variable into one hashmap:
        (apply merge)))
 
-(defn- wrap-raw-body-rdr
-  "Create a Reader corresponding to the raw request body and add it to the request."
+(defn- wrap-body-bytes
+  "Copy the bytes of the request body to :body-bytes."
   [handler]
-  ;; Note that this code is adapted from: https://stackoverflow.com/a/37397991
+  ;; Note that this code is adapted from: https://stackoverflow.com/a/20570399
   (fn [request]
-    (let [raw-body (body-string request)]
-      (->> raw-body (char-array) (io/reader) (assoc request :raw-body-rdr) (handler)))))
+    (let [buffer (ByteArrayOutputStream.)
+          _ (when (:body request) (io/copy (:body request) buffer))
+          bytes (.toByteArray buffer)]
+      (handler
+       (assoc
+        request
+        :body (ByteArrayInputStream. bytes)
```

Comment:
Just wondering, why have both `:body` and `:body-bytes`? (why not just `:body-bytes`?)

Reply:
I think that `:body` needs to be read again by other middleware, but I’ll double check that.